File: scripts/number_publisher.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64


cont_pub = Int64()
def pub_num():
    pub = rospy.Publisher('number', Int64, queue_size=1)
    rospy.init_node('number_publisher')
    rate = rospy.Rate(2) # 2hz
    cont=0
    while not rospy.is_shutdown():
	cont=cont+1
	cont_pub.data=cont
	rospy.loginfo(cont)
        pub.publish(cont_pub)
        rate.sleep()

if __name__ == '__main__':
	try:
		pub_num()
	except rospy.ROSInterruptException:
		pass

File: src/number_publisher.cpp
#include "ros/ros.h"
#include "std_msgs/Int64.h"



int main(int argc, char **argv)
{
  ros::init(argc, argv, "number_publisher");
  ros::NodeHandle n;
  ros::Publisher pub = n.advertise<std_msgs::Int64>("number", 1);
  ros::Rate loop_rate(2);

  int count = 0;
  while (ros::ok())
  {
    std_msgs::Int64 msg;
    msg.data= count;
    ROS_INFO("Enviado %i", msg.data);
    pub.publish(msg);
    loop_rate.sleep();
    ++count;
  }

  return 0;
}
